lib/internal/event_target.js

```diff
@@ -0,0 +1,392 @@
+'use strict';
+
+const {
+  ArrayFrom,
+  Object,
+  Map,
+  SafeWeakMap,
+  Set,
+  SymbolFor,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_THIS,
+    ERR_EVENT_RECURSION,
+  }
+} = require('internal/errors');
+const kRejection = SymbolFor('nodejs.rejection');
+
+const perf_hooks = require('perf_hooks');
+const { customInspectSymbol } = require('internal/util');
+const { inspect } = require('util');
+
+const events = new SafeWeakMap();
+const targets = new SafeWeakMap();
+
+class Event {
+  constructor(type, options) {
+    if (typeof type !== 'string')
+      throw new ERR_INVALID_ARG_TYPE('type', 'string', type);
+    if (options != null && typeof options !== 'object')
+      throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
+    const { cancelable } = { ...options };
```

Comment:
The DOM API deals with things like passing ` ({ get cancelable() { throw new Error(); } })`, I'm not saying we should pass the WPT - but if we don't expose `Event` to begin with as part of our API surface (and only export EventTarget for AbortController and streams) it's a whole class of issues we don't have to deal with (yet!)

Reply:
(Actually: reading the spec I think passing a throwing getter just throws here so that part is fine)